src/main/java/com/cloudbees/jenkins/GitHubPushCause.java

```diff
@@ -34,4 +34,26 @@ public String getShortDescription() {
         String pusher = pushedBy != null ? pushedBy : "";
         return "Started by GitHub push by " + pusher;
     }
+
+    @Override
+    public int hashCode() {
+        int hash = 5;
+        hash = 79 * hash + (this.pushedBy != null ? this.pushedBy.hashCode() : 0);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final GitHubPushCause other = (GitHubPushCause) obj;
```

Comment:
`getClass().isInstance(obj)`
instead of
```
if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }

```